website/content/api-docs/auth/aws.mdx

```diff
@@ -65,6 +65,10 @@ capabilities, the credentials are fetched automatically.
 - `sts_region` `(string: "")` - Region to override the default region for making
   AWS STS API calls. Should only be set if `sts_endpoint` is set. If so, should
   be set to the region in which the custom `sts_endpoint` resides.
+- `sts_region_from_client` `(boolean: false)` - If set, overrides both `sts_endpoint`
+  and `sts_region` to instead use the region specified in the client request headers for
+  IAM-based authentication . This can be useful when you have client requests coming from
+  different regions and want flexibility in which regional STS API is used.
```

Comment:
Is it more typical to override other parameters or to enforce mutual exclusivity on the API?

I think for this particular use case going with an override sounds intuitive, but I'm wondering if Vault has an established preference that users may expect.

Reply:
That's a good callout. From what I can tell, overriding other parameters to change behavior doesn't seem to be a new concept in Vault. And searching for "override" in the codebase yields a more than a few results. It does seem to be a natural evolution of the API :)

Reply:
Gotcha, thanks for checking!